src/Barbies/Internal/DistributiveB.hs

```diff
@@ -35,9 +35,9 @@ import Data.Kind              (Type)
 --  instances need to satisfy these laws:
 --
 -- @
--- 'bdistribute' . h = 'bmap' ('Compose' . h . 'getCompose') . 'bdistribute'    -- naturality
+-- 'bdistribute' . fmap (bmap h) = 'bmap' ('Compose' . fmap h . 'getCompose') . 'bdistribute'    -- naturality
```

Comment:
I'm not sure about this one. The naturality condition for `sequenceA` is:

```haskell
h . sequenceA @g = sequenceA @g . fmap @g h -- for every h :: forall a . f1 a -> f2 a
```
So, I think the condition for `distribute` then becomes:

```haskell
distribute @g @f2 . h = fmap @g h . distribute @g @f1  -- for every h :: forall a . f1 a -> f2 a
```

And the higher-order version needs to take into account the `Compose`:

```haskell
bdistribute @b @f2 . h = bmap @b (Compose . h . getCompose) . bdistribute @b @f1  -- for every h :: forall a . f1 a -> f2 a
```

Reply:
You're right of course, I was too quick to give up on typing the law for `test/Spec/Distributive.hs`.